src/crimson/net/SocketMessenger.cc

```diff
@@ -154,6 +154,9 @@ seastar::future<> SocketMessenger::shutdown()
 {
   assert(seastar::this_shard_id() == master_sid);
   return seastar::futurize_invoke([this] {
+    if (dispatchers) {
+      assert(dispatchers->empty());
+    }
```

Comment:
I think, in our scenarios, all dispatchers must have themselves removed from messengers before the shutting down of messengers, otherwise things may go wrong. So I think, we need to assert it here, to insure future development won't violate that rule.